src/lxd.py

```diff
@@ -0,0 +1,321 @@
+
+
+class LxdInstanceFiles:
+    """File manager of a LXD instance.
+
+    Attrs:
+        instance (LxdInstance): LXD instance where the files are located in.
+    """
+
+    def __init__(self, instance: LxdInstance):
+        """Construct the file manager.
+
+        Args:
+            instance: LXD instance where the files are located in.
+        """
+        self.instance = instance
+
+    def mk_dir(self, dir_name: str) -> None:
+        """Create a directory in the LXD instance.
+
+        Args:
+            dir: Name of the directory to create.
+        """
+        self.instance.execute(["/usr/bin/mkdir", "-p", dir_name])
+
+    def push(self, source: str, destination: str, mode: Optional[str] = None) -> None:
+        """Push a file to the LXD instance.
+
+        Args:
+            source: Path of the file to push to the LXD instance.
+            destination: Path in the LXD instance to load the file.
+            mode: File permission setting.
+
+        Raises:
+            LxdException: Unable to load the file into the LXD instance.
+        """
+        lxc_cmd = [
+            "/snap/bin/lxc",
+            "file",
+            "push",
+            source,
+            f"{self.instance.name}/{destination.lstrip('/')}",
+        ]
+
+        if mode:
+            lxc_cmd += ["--mode", str(mode)]
+
+        try:
+            execute_command(lxc_cmd)
+        except CalledProcessError as err:
+            raise LxdError(f"Unable to push file into LXD instance {self.instance.name}") from err
+
+    def put(self, filepath: str, content: Union[str, bytes], mode: Optional[str] = None) -> None:
+        """Put a file with the given content in the LXD instance.
+
+        Args:
+            filepath: Path in the LXD instance to load the file.
+            content: Content of the file.
+            mode: File permission setting.
+
+        Raises:
+            LxdException: Unable to load the file to the LXD instance.
+        """
+        if isinstance(content, str):
+            content = content.encode("utf-8")
+
+        with tempfile.NamedTemporaryFile() as file:
+            file.write(content)
+            file.flush()
+
+            self.push(file.name, filepath, mode)
+
+    def pull(self, source: str, destination: str) -> None:
+        """Pull a file from the LXD instance.
+
+        Args:
+            source: Path of the file to pull in the LXD instance.
+            destination: Path of load the file.
+
+        Raises:
+            LxdException: Unable to load the file from the LXD instance.
+        """
+        lxc_cmd = [
+            "/snap/bin/lxc",
+            "file",
+            "pull",
+            f"{self.instance.name}/{source.lstrip('/')}",
+            destination,
+        ]
+
+        try:
+            execute_command(lxc_cmd)
+        except CalledProcessError as err:
+            raise LxdError(
+                f"Unable to pull file {str(source)} from LXD instance {self.instance.name}"
```

Comment:
```suggestion
                f"Unable to pull file {source} from LXD instance {self.instance.name}"
```
The str conversion can be omitted :)

Reply:
There is already a comment addressing this, resolving, sorry about the duplicated issue!